agent/spec/lib/kontena/workers/event_worker_spec.rb

```diff
@@ -44,7 +46,7 @@
         times.times {
           block.call(Docker::Event.new(event))
         }
-        sleep 0.1
+        sleep 0.5
```

Comment:
> The streams and processes events spec is also racy as in #2321 (comment), but fixing that would need changes to the EventWorker, which I'd rather avoid right now.

Alternatively just bump the `sleep` to avoid the race condition. With a 10ms sleep, I'm getting about 40-50/100 events on my laptop... 20ms still fails every time at 80-90, but 50ms mostly passes but still fails occasionally.  It's currently at 100ms, so if we bump that to say 500ms, then it should be enough to make the probably of travis spec failures rare enough to not be an issue that we need to care about anymore. 

The entire `stop_processing` mechanism is flawed, but it's probably not worth fixing because it's only really used in the specs... it's also used in the actor finalizer, but the only thing it really serves to do there is to suppress any error logging... but I wouldn't except any of the `!processing?` error cases to necessarily even get hit during the agent shutdown process, because AFAIK the `Docker::Event.stream` in the `Celluloid::Future` thread won't get any signal leading to a ruby exception during shutdown?